agent/testutils/fakes.go

```diff
@@ -0,0 +1,219 @@
+package testutils
+
+import (
+	"net"
+	"sync"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"github.com/docker/swarmkit/agent/exec"
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/ca"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/context"
+)
+
+// TestExecutor is executor for integration tests
+type TestExecutor struct {
```

Comment:
Note:  `TestExecutor` and `TestController` are just the functions from `integration/exec.go`, modified so that we can inject arbitrary node descriptions.